### PR TITLE
html: Add a testcase for submit event reentrancy protection

### PR DESCRIPTION
**Pull request by tkent-google**
Specification: https://github.com/whatwg/html/pull/4621

### CODE REVIEW

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html [hunk 1]
```diff
@@ -18,16 +18,33 @@
...
+    hidden.value = 'v2';
+  };
+  submitter.click();
+  iframe.onload = t.step_func_done(() => {
+    assert_not_equals(iframe.contentWindow.location.search.indexOf('n=v2'), -1);
+  });
+}, "If form's firing submit event is true, then return");
+
+let test9 = async_test('Cannot navigate (after constructing the entry list)');
+test9.step(() => {
```
Comment (domenic) [inline html/semantics/forms/form-submission-0/form-submission-algorithm.html]: Was there a reason to change this from test10 to test9? Seems to just make the diff a bit more complicated.

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html [hunk 2]
```diff
@@ -18,16 +18,33 @@
   assert_equals(counter, 2);
 }, 'If constructing entry list flag of form is true, then return');
 
-let test10 = async_test('Cannot navigate (after constructing the entry list)');
-test10.step(() => {
+
+async_test(t => {
```
Comment (domenic) [inline html/semantics/forms/form-submission-0/form-submission-algorithm.html]: Do you think this style of test is better than the counter style used above? If so, maybe add a comment?

**Review by domenic — APPROVED** [review]
LGTM with minor question/nit

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html (hunk 1 above)
Comment (tkent-google) [reply html/semantics/forms/form-submission-0/form-submission-algorithm.html]: '10' was the step number in 'form submission algorithm', and this test case tests that step.  The specification PR changed the step number :)

Updating the step number every time won't work well.  I changed this to async_test(t => {}, 'Cannot navigate (...)').

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html [hunk 3]
```diff
@@ -19,6 +19,22 @@
 }, 'If constructing entry list flag of form is true, then return');
 
 
+test(() => {
+  let form = populateForm('<input required><input type=submit><button type=submit></button>');
+  let submitter1 = form.querySelector('input[type=submit]');
+  let submitter2 = form.querySelector('button[type=submit]');
+  let iframe = form.previousSibling;
```
Comment (domenic) [inline html/semantics/forms/form-submission-0/form-submission-algorithm.html]: iframe reference seems unused?

**Review by domenic — APPROVED** [review]
New test LGTM with nit, although still unclear why one test uses navigation and the others use counters.

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html (hunk 2 above)
Comment (tkent-google) [reply html/semantics/forms/form-submission-0/form-submission-algorithm.html]: I'd like to test which one of two ```click()```s submits the form.  I'll add a comment.

File: html/semantics/forms/form-submission-0/form-submission-algorithm.html (hunk 3 above)
Comment (tkent-google) [reply html/semantics/forms/form-submission-0/form-submission-algorithm.html]: Removed.

**Pull request merged**